### ISSUE TITLE
`OpenOptions` with append & truncate enabled: misleading error message

### ISSUE THREAD

**Issue by Radrogyl** [rust-lang/rust#160716]
When `OpenOptions` have both `append(true)` and `truncate(true)` set, the error message is misleading/unhelpful. Currently the message is as follows:
`Custom { kind: InvalidInput, error: "creating or truncating a file requires write or append access" }`

Both append and truncate don’t mention of being exclusive options.

I experienced this error with the following code. Note that `create(true)` and `write(true)` both don’t change the behavior, any combination where both append and truncate are set results in the error message above.

```rust
let file_out = File::options()
        .create(true)
        .write(true)
        .append(true)
        .truncate(true)
        .open("./data/some_file")
        .unwrap();
```

rustc --version `rustc 1.97.1 (8bab26f4f 2026-07-14)`
